### PR TITLE
Fix usage of safe math

### LINKED ISSUE #419: bug: Fix usage of SafeMath in arithmetic opcodes
# Bug Report

**Current behavior:**

The PR https://github.com/sayajin-labs/kakarot/pull/415 fixed usage of `SafeMath` for `ADD` and `MUL`. Some others opcodes still use `SafeMath`.

**Expected behavior:**

The goal of this issue is to find out wether we need to keep them or not by looking at the EVM spec.

**Related code:**

see `src/kakarot/instructions/stop_and_arithmetic_operations.cairo`

### PR DESCRIPTION
**Pull request by FranFiuba**
Replaced SafeMath arithmetic operations with uint256 operations. 

I spent two days, it was my first Cairo and EVM pr and used to understand some details. 

## Pull request type

Please check the type of change your PR introduces:

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

The PR [#413](https://github.com/sayajin-labs/kakarot/pull/415) fixed the usage of SafeMath  on Multiplication and Addition, the rest of the arithmetic operations are still using SafeMath. 

Resolves #419

## What is the new behavior?

All arithmetic operations now use  [uint256](https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/cairo/common/uint256.cairo). 

## Other information

According with the specs on the EVM we should'n use safe mode. The EVM use "wrap-around" arithmetic, which means that overflow and underflow conditions are not detected or handled in the same way as in traditional computer architectures. Instead, the EVM wraps around to the opposite end of the range of possible values.

In summary if we use SafeMath we would have incompatibility issues.

### CODE REVIEW

**Comment by Eikix** [comment]
Could you comment on #419 and ask for assignment there? That way it is clear that the issue is being taken care of and we avoid duplicated work

In addition, the PR template has a "resolves #" field so that you may show which issue your PR solves

If you have any questions, please ask on the telegram chat https://t.me/KakarotZkEvm

**Review by ClementWalter — APPROVED** [review]
